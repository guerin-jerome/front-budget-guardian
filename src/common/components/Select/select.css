select {
  padding: 0 35px 0 10px;
  margin-top: 5px;
  height: 38px;
  width: 100%;
  font-size: 14.5px;
  font-family: Arial, sans-serif;
  border-radius: 2px;
  border: 1px solid #ccc;

  /* Cacher la flèche par défaut */
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;

  /* Ajouter une image de fond personnalisée pour la flèche */
  background-image: url("./assets/arrow-down.svg");
  background-repeat: no-repeat;
  background-position: right center;
  background-size: auto 50%;
  background-position-x: calc(100% - 10px);
}
