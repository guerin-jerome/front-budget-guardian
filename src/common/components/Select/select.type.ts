import { UseFormRegisterReturn } from "react-hook-form";

export type SelectOption = {
  value: string;
  text: string;
  disabled?: boolean;
};

export type SelectProps = {
  options: SelectOption[];
  hasError: boolean;
  register?: UseFormRegisterReturn;
  defaultValue?: string;
  defaultText?: string;
};
