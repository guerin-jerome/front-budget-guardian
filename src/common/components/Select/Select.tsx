import { SelectProps } from "./select.type";
import "./select.css";

export const Select = ({
  options,
  register,
  defaultValue,
  defaultText = "SÃ©lectionnez une valeur",
}: SelectProps) => {
  const placeholderSelect = {
    value: "initial_value",
    text: defaultText,
    disabled: true,
  };

  return (
    <select {...register} defaultValue={defaultValue}>
      {[placeholderSelect, ...options].map(({ value, text, disabled }) => (
        <option key={value} value={value} disabled={disabled}>
          {text}
        </option>
      ))}
    </select>
  );
};
