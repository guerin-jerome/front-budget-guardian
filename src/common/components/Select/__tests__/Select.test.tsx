import { render, screen } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { Select } from "../Select";

describe("<Select />", () => {
  it("matchs snapshot", () => {
    const { container } = render(<Select options={[]} />);
    expect(container).toMatchSnapshot();
  });

  it("renders well", async () => {
    render(
      <Select
        defaultValue="1"
        defaultText="Example"
        options={[{ value: "1", text: "optionTest", disabled: true }]}
      />
    );
    const select = screen.getByRole("combobox");
    const placeholerOption = screen.getByRole("option", { name: "Example" });
    const optionTest = screen.getByRole("option", { name: "optionTest" });

    expect(select).toBeDefined();
    expect(placeholerOption).toBeDefined();
    expect(optionTest).toBeDefined();
  });
});
