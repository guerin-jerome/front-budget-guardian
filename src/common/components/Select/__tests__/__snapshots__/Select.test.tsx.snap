// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Select /> > matchs snapshot 1`] = `
<div>
  <select
    id="test"
    name="test"
  />
</div>
`;
